### PR TITLE
Update to Errorprone 2.7

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java [hunk 1]
```diff
@@ -74,4 +75,16 @@ public boolean equals(Object obj) {
   public int hashCode() {
     return Objects.hash(connectionContext, clientParentContext, context);
   }
+
+  @Override
+  public String toString() {
```
Comment (mateuszrzeszutek) [inline instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:80]: Is that `toString()` needed?

File: instrumentation/vertx-reactive-3.5/javaagent/src/test/java/VertxReactiveWebServer.java [hunk 2]
```diff
@@ -126,7 +126,6 @@ private static void handleListProducts(RoutingContext routingContext) {
           .rxQuery("SELECT id" + queryInfix + ", name, price, weight FROM products")
           .flatMap(
               result -> {
-                Thread.dumpStack();
```
Comment (mateuszrzeszutek) [inline instrumentation/vertx-reactive-3.5/javaagent/src/test/java/VertxReactiveWebServer.java:129]: 😱

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java (hunk 1 above)
Comment (anuraaga) [reply instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:80]: There's a check that toString is added if equals/hashcode are. I can disable it if it seems to much but figure having a toString is generally better than not.

Comment (mateuszrzeszutek) [reply instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:80]: Hmm, it's not really a big deal or anything, but those methods will probably be never called - they're just dead code. If we need a `toString()` method we usually add it.

Comment (trask) [reply instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:80]: I was going to say +1 for disabling this rule, then went to read why the rule exists, and didn't realize `Object.toString()` was affected by overriding `hashCode()` 😱, I guess I thought it was always based on `System.identityHashCode()`, so now at least the rule makes sense to me. I think I'm probably still _slightly_ +1 for disabling the rule for the same reasons as @mateuszrzeszutek mentioned, but also good with keeping it if you think it's useful.

**Review by trask — APPROVED** [review]
nice

File: instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java (hunk 1 above)
Comment (anuraaga) [reply instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:80]: IntelliJ calls toString when debugging which is the only reason really to have toString on types that won't be called by code. Not that debugging works well for us :) But it's why I slightly lean towrads keeping it

**Pull request merged**